### PR TITLE
#155 | The infrastructure for Pagination, Order, and Filter has been created.

### PR DESCRIPTION
**Pull request by MenekseYuncu**
## Release Notes

- **New Features**
	- Introduced `BaseFilter`, `BasePage`, `BasePageable`, and `BaseSort` classes to enhance pagination, filtering, and sorting capabilities.
	- Added methods for converting filter criteria into specifications and managing paginated data.
	- Enhanced `PagingRequest` and `PagingResponse` classes to support new pagination and filtering structures.
	- Added `AdminPostFilter` for filtering posts based on active status.
	- Introduced `TicketMessageFilter` for filtering ticket messages based on their ID.

- **Bug Fixes**
	- Updated method signatures across services to utilize the new pagination model, ensuring consistency in data retrieval.

- **Refactor**
	- Refactored various classes to replace the deprecated `Paging` class with the new `BasePage` class, improving the overall structure and functionality of pagination handling.
	- Modified `AdminController`, `PostService`, and `PostServiceImpl` to accommodate the new pagination model.
	- Updated `TicketMessageRequest` to integrate the new `TicketMessageFilter`.

### CODE REVIEW

File: src/main/java/org/gelecekbilimde/scienceplatform/common/model/BasePageable.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import com.nimbusds.oauth2.sdk.util.CollectionUtils;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.experimental.SuperBuilder;
+import org.hibernate.validator.constraints.Range;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
```
[the remark this answers is not in the document]
Comment (MenekseYuncu) [reply src/main/java/org/gelecekbilimde/scienceplatform/common/model/BasePageable.java:13]: @coderabbitai import java.util.Collections does not include CollectionUtils

File: src/main/java/org/gelecekbilimde/scienceplatform/ticket/service/impl/TicketMessageServiceImpl.java [hunk 2]
```diff
@@ -44,9 +44,9 @@ public PagingResponse<TicketMessageResponse> ticketMessageRead(TicketMessageRequ
 	@Override
 	public PagingResponse<TicketMessageResponse> ticketMessageReadSelf(TicketMessageRequest request) {
 		Specification<TicketMessageEntity> spec = (root, query, criteriaBuilder) -> criteriaBuilder.equal(root.get("userId"), identity.getUserId());
-		Page<TicketMessageEntity> messagePage = ticketMessageRepository.findAll(spec, request.toPageable());
+		Page<TicketMessageEntity> messagePage = ticketMessageRepository.findAll(spec, request.getPageable().toPageable());
 		List<TicketMessageResponse> ticketMessageRespons = ticketMessageEntityToMessageResponseMapper.map(messagePage.getContent());
-		final Paging<TicketMessageResponse> messageResponsesPage = Paging.of(messagePage, ticketMessageRespons);
+		final BasePage<TicketMessageResponse> messageResponsesPage = BasePage.of(messagePage, ticketMessageRespons);
```
[the remark this answers is not in the document]
Comment (MenekseYuncu) [reply src/main/java/org/gelecekbilimde/scienceplatform/ticket/service/impl/TicketMessageServiceImpl.java:49]: @coderabbitai This part will be corrected later.

File: src/main/java/org/gelecekbilimde/scienceplatform/common/model/BasePageable.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+@NoArgsConstructor
+public class BasePageable extends BaseSort {
+
+
+	@Range(min = 1, max = 99999999)
+	private int page;
+
+
+	@Range(min = 1, max = 99999999)
+	private int pageSize;
```
[the remark this answers is not in the document]
Comment (MenekseYuncu) [reply src/main/java/org/gelecekbilimde/scienceplatform/common/model/BasePageable.java:27]: @aleyna-yildizli @Ciklamert page ve pageSize max kaç olmalı ? ben max = 99999999 verdim ama ne kadar doğru bilemedim

File: src/main/java/org/gelecekbilimde/scienceplatform/common/model/BasePageable.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+@AllArgsConstructor
+@NoArgsConstructor
+public class BasePageable extends BaseSort {
+
+
+	@Range(min = 1, max = 99999999)
+	private int page;
+
+
+	@Range(min = 1, max = 10)
```
Comment (agitrubard) [inline src/main/java/org/gelecekbilimde/scienceplatform/common/model/BasePageable.java:26]: Şimdilik 10 olarak sabitleyelim diye düşünüyorum. Sonra duruma göre dinamikleştirilirse düzeltelim ✅ 
```suggestion
	@Range(min = 10, max = 10)
```

File: src/main/java/org/gelecekbilimde/scienceplatform/post/model/request/AdminPostListRequest.java [hunk 5]
```diff
@@ -1,13 +1,25 @@
...
 import lombok.EqualsAndHashCode;
 import org.gelecekbilimde.scienceplatform.common.model.request.PagingRequest;
 
+import java.util.Set;
+
 @EqualsAndHashCode(callSuper = true)
 @Data
 public class AdminPostListRequest extends PagingRequest {
 
 	public Boolean isActive;
```
Comment (agitrubard) [inline src/main/java/org/gelecekbilimde/scienceplatform/post/model/request/AdminPostListRequest.java:15]: Bu alanı Filter içerisine kapsülleyebilir miyiz? Bu sayede yapı görünmüş olur ✅

File: src/main/java/org/gelecekbilimde/scienceplatform/ticket/model/request/TicketMessageRequest.java [hunk 6]
```diff
@@ -5,11 +5,18 @@
...
 import org.gelecekbilimde.scienceplatform.common.model.request.PagingRequest;
 
+import java.util.Set;
+
 @Getter
 @Setter
 public class TicketMessageRequest extends PagingRequest {
 
 	@NotNull
 	private Integer id;
```
Comment (agitrubard) [inline src/main/java/org/gelecekbilimde/scienceplatform/ticket/model/request/TicketMessageRequest.java:15]: Bu alanı Filter içerisine kapsülleyebilir miyiz? Bu sayede yapı görünmüş olur ✅

**Review by agitrubard — CHANGES_REQUESTED** [review]

**Review by agitrubard — APPROVED** [review]

**Pull request merged**